backbone.js

```diff
@@ -1126,16 +1126,21 @@
   // Underscore methods that we want to implement on the Collection.
   // 90% of the core usefulness of Backbone Collections is actually implemented
   // right here:
-  var collectionMethods = { forEach: 3, each: 3, map: 3, collect: 3, reduce: 4,
-      foldl: 4, inject: 4, reduceRight: 4, foldr: 4, find: 3, detect: 3, filter: 3,
-      select: 3, reject: 3, every: 3, all: 3, some: 3, any: 3, include: 2,
-      contains: 2, invoke: 0, max: 3, min: 3, toArray: 1, size: 1, first: 3,
-      head: 3, take: 3, initial: 3, rest: 3, tail: 3, drop: 3, last: 3,
-      without: 0, difference: 0, indexOf: 3, shuffle: 1, lastIndexOf: 3,
-      isEmpty: 1, chain: 1, sample: 3, partition: 3 };
+  var collectionMethods = { forEach: 3, each: 3, map: 3, collect: 3, reduce: 4, foldl: 4,
+        inject: 4, reduceRight: 4, foldr: 4, include: 2, contains: 2, invoke: 0,
+        max: 3, min: 3, toArray: 1, size: 1, first: 3, head: 3, take: 3, initial: 3,
+        rest: 3, tail: 3, drop: 3, last: 3, without: 0, difference: 0, indexOf: 3,
+        shuffle: 1, lastIndexOf: 3, isEmpty: 1, chain: 1, sample: 3 },
+      collectionMethodsWithModelMatching = { find: 3, detect: 3, filter: 3, select: 3,
+        reject: 3, every: 3, all: 3, any: 3, some: 3, partition: 3 };
```

Comment:
I think it can only apply to filtering methods.

What should return ```collection.map({user: 'lechuck'})```? (not filter, but transformation)
And ```collection.max({user: 'lechuck'})```? (max must take a compared attribute, not a set of attributes)